[BUG] Compilation errors occured if set multipart/type in requestBody

#### Bug Report Checklist

- [ v] Have you provided a full/minimal spec to reproduce the issue?
- [ v] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [ v] Have you searched for related issues/PRs?
- [ v] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

Compilation errors occured if set multipart/type in requestBody:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project drones-openfeign-api: Compilation failure: Compilation failure: 
[ERROR] /opanapicheck/drones-openfeign-api/target/generated-sources/openapi/src/main/java/com/tech/drones/oa/api/FacadeApi.java:[8,32] package com.tech.drones.oa.model does not exist
[ERROR] /opanapicheck/drones-openfeign-api/target/generated-sources/openapi/src/main/java/com/tech/drones/oa/api/FacadeApi.java:[62,25] cannot find symbol
[ERROR]   symbol:   class Medication
[ERROR]   location: interface com.tech.drones.oa.api.FacadeApi
[ERROR] /opanapicheck/drones-openfeign-api/target/generated-sources/openapi/src/main/java/com/tech/drones/oa/api/FacadeApi.java:[51,95] cannot find symbol
[ERROR]   symbol:   class Medication
[ERROR]   location: interface com.tech.drones.oa.api.FacadeApi
[ERROR] -> [Help 1]
```

##### openapi-generator version

6.2.1

##### OpenAPI declaration file content or url

https://gist.github.com/tbw777/d74e13d3da95b2b314c0d338a66dfa46

##### Generation Details

https://gist.github.com/tbw777/18f0077e8c3da0dc3834a2f836fe1006

##### Steps to reproduce

mvn clean compile

##### Related issues/PRs

not found

##### Suggest a fix

dont have

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java
@@ -6905,6 +6905,13 @@ public class DefaultCodegen implements CodegenConfig {
         LOGGER.debug("debugging fromRequestBodyToFormParameters= {}", body);
         Schema schema = ModelUtils.getSchemaFromRequestBody(body);
         schema = ModelUtils.getReferencedSchema(this.openAPI, schema);
+        if(ModelUtils.isMapSchema(schema)) {
+            LOGGER.error("Form parameters with additionalProperties are not supported by OpenAPI Generator. Please report the issue to https://github.com/openapitools/openapi-generator if you need help.");
+        }
+        if(ModelUtils.isArraySchema(schema)) {
+            LOGGER.error("Array form parameters are not supported by OpenAPI Generator. Please report the issue to https://github.com/openapitools/openapi-generator if you need help.");
+        }
+
         List<String> allRequired = new ArrayList<>();
         Map<String, Schema> properties = new LinkedHashMap<>();
         // this traverses a composed schema and extracts all properties in each schema into properties
@@ -6920,10 +6927,6 @@ public class DefaultCodegen implements CodegenConfig {
                 // value => property schema
                 String propertyName = entry.getKey();
                 Schema propertySchema = entry.getValue();
-                if (ModelUtils.isMapSchema(propertySchema)) {
-                    LOGGER.error("Map of form parameters not supported. Please report the issue to https://github.com/openapitools/openapi-generator if you need help.");
-                    continue;
-                }
                 codegenParameter = fromFormProperty(propertyName, propertySchema, imports);
 
                 // Set 'required' flag defined in the schema element

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java
@@ -367,7 +367,7 @@ public class ModelUtils {
         Map<String, Schema> properties = schema.getProperties();
         if (properties != null) {
             for (Schema property : properties.values()) {
-                visitSchema(openAPI, property, mimeType, visitedSchemas, visitor);
+                visitSchema(openAPI, property, null, visitedSchemas, visitor);
             }
         }
     }

--- a/modules/openapi-generator/src/test/java/org/openapitools/codegen/utils/ModelUtilsTest.java
+++ b/modules/openapi-generator/src/test/java/org/openapitools/codegen/utils/ModelUtilsTest.java
@@ -114,6 +114,14 @@ public class ModelUtilsTest {
     }
 
     @Test
+    public void testNestedFormParameter() {
+        final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/2_0/nestedFormParameter.yaml");
+        List<String> unusedSchemas = ModelUtils.getSchemasUsedOnlyInFormParam(openAPI);
+        Assert.assertEquals(unusedSchemas.size(), 1);
+        Assert.assertTrue(unusedSchemas.contains("OuterObject"), "contains 'OuterObject'");
+    }
+
+    @Test
     public void testNoComponentsSection() {
         final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_0/ping.yaml");
         List<String> unusedSchemas = ModelUtils.getSchemasUsedOnlyInFormParam(openAPI);

--- a/modules/openapi-generator/src/test/resources/2_0/nestedFormParameter.yaml
+++ b/modules/openapi-generator/src/test/resources/2_0/nestedFormParameter.yaml
@@ -1,0 +1,28 @@
+swagger: "2.0"
+info:
+  title: dummy
+  version: dummy
+paths:
+  /property:
+    post:
+      consumes:
+        - application/x-www-form-urlencoded
+      parameters:
+        - in: body
+          name: body
+          schema:
+              $ref: '#/definitions/OuterObject'
+      responses:
+        '200':
+          description: dummy
+definitions:
+  InnerObject:
+    type: object
+    properties:
+      innerProperty:
+        type: string
+  OuterObject:
+    type: object
+    properties:
+      outerProperty:
+        $ref: '#/definitions/InnerObject'

